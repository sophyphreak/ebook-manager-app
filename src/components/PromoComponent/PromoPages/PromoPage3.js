import React, { Component } from 'react';
import {
  Card,
  CardHeader,
  CardBody,
  Button,
  Form
} from 'reactstrap';

import Description from '../../FormElements/Description';
import AuthorBio from '../../FormElements/AuthorBio';

const PromoPage3 = ({
  error,
  onBack,
  onSubmit,

  description,
  onDescriptionChange,

  authorBio,
  onAuthorBioChange
}) => (
    <Card className="animated fadeIn">
      <CardHeader>
        <h4><strong>Book Promo - 3 of 3</strong></h4>
      </CardHeader>
      <CardBody>
        <Form onSubmit={onSubmit}>
          <Description
            value={description}
            onChange={onDescriptionChange}
          />
          <AuthorBio
            value={authorBio}
            onChange={onAuthorBioChange}
          />
          {error && <p>{error}</p>}
          <Button
            onClick={onBack}
            size="sm"
            color="basic"
          >
            Go Back
          </Button>
          <Button
            type="submit"
            size="sm"
            color="primary"
            value="Submit"
          >
            Next Section
          </Button>
        </Form>
      </CardBody>
    </Card>
  );

export default PromoPage3;