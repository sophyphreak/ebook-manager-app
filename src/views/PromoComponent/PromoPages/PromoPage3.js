import React, { Component } from 'react';
import { Card, CardHeader, CardBody, Button, Form } from 'reactstrap';

import Description from '../../FormElements/Description';
import AuthorBio from '../../FormElements/AuthorBio';

const PromoPage3 = ({
  error,
  onBack,
  onSubmit,

  description,
  onDescriptionChange,

  authorBio,
  onAuthorBioChange
}) => (
  <Card className="animated fadeIn">
    <CardHeader>
      <h4>
        <strong>Book Promo - 3 of 3</strong>
      </h4>
    </CardHeader>
    <CardBody>
      <Form onSubmit={onSubmit}>
        <Description
          label="Promotional Book Description (Max 2,000 characters)*"
          description={description}
          onDescriptionChange={onDescriptionChange}
          hasError={!!error.message}
          errorMessage={error.description}
        />
        <AuthorBio
          label="Author Biography:*"
          authorBio={authorBio}
          onAuthorBioChange={onAuthorBioChange}
          hasError={!!error.message}
          errorMessage={error.authorBio}
        />
        {error.message && <p>{error.message}</p>}
        <Button
          className=".back-button"
          onClick={onBack}
          size="sm"
          color="basic"
        >
          Go Back
        </Button>
        <Button type="submit" size="sm" color="primary" value="Submit">
          Submit
        </Button>
      </Form>
    </CardBody>
  </Card>
);

export default PromoPage3;
