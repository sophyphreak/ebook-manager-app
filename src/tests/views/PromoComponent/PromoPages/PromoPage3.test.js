import React from 'react';
import { shallow } from 'enzyme';
import PromoPage3 from '../../../../views/PromoComponent/PromoPages/PromoPage3';

let error, wrapper;

beforeEach(() => {
  error = {
    message: '',
    description: '',
    authorBio: ''
  };
  wrapper = shallow(<PromoPage3 error={error} />);
});

test('should render PromoPage3 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
