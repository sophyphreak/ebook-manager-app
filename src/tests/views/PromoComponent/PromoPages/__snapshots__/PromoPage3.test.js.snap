// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PromoPage3 correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PromoPage3
    error={
      Object {
        "authorBio": "",
        "description": "",
        "message": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardHeader
          tag="div"
        >
          <h4>
            <strong>
              Book Promo - 3 of 3
            </strong>
          </h4>
        </CardHeader>,
        <CardBody
          tag="div"
        >
          <Form
            tag="form"
          >
            <Description
              errorMessage=""
              hasError={false}
              label="Promotional Book Description (Max 2,000 characters)*"
            />
            <AuthorBio
              errorMessage=""
              hasError={false}
              label="Author Biography:*"
            />
            
            <Button
              className=".back-button"
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Submit
            </Button>
          </Form>
        </CardBody>,
      ],
      "className": "animated fadeIn",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h4>
            <strong>
              Book Promo - 3 of 3
            </strong>
          </h4>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Book Promo - 3 of 3
            </strong>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Book Promo - 3 of 3",
            },
            "ref": null,
            "rendered": "Book Promo - 3 of 3",
            "type": "strong",
          },
          "type": "h4",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Form
            tag="form"
          >
            <Description
              errorMessage=""
              hasError={false}
              label="Promotional Book Description (Max 2,000 characters)*"
            />
            <AuthorBio
              errorMessage=""
              hasError={false}
              label="Author Biography:*"
            />
            
            <Button
              className=".back-button"
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Submit
            </Button>
          </Form>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Description
                errorMessage=""
                hasError={false}
                label="Promotional Book Description (Max 2,000 characters)*"
              />,
              <AuthorBio
                errorMessage=""
                hasError={false}
                label="Author Biography:*"
              />,
              "",
              <Button
                className=".back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>,
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>,
            ],
            "onSubmit": undefined,
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": undefined,
                "errorMessage": "",
                "hasError": false,
                "label": "Promotional Book Description (Max 2,000 characters)*",
                "onDescriptionChange": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "authorBio": undefined,
                "errorMessage": "",
                "hasError": false,
                "label": "Author Biography:*",
                "onAuthorBioChange": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go Back",
                "className": ".back-button",
                "color": "basic",
                "onClick": undefined,
                "size": "sm",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Go Back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Submit",
                "color": "primary",
                "size": "sm",
                "tag": "button",
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            tag="div"
          >
            <h4>
              <strong>
                Book Promo - 3 of 3
              </strong>
            </h4>
          </CardHeader>,
          <CardBody
            tag="div"
          >
            <Form
              tag="form"
            >
              <Description
                errorMessage=""
                hasError={false}
                label="Promotional Book Description (Max 2,000 characters)*"
              />
              <AuthorBio
                errorMessage=""
                hasError={false}
                label="Author Biography:*"
              />
              
              <Button
                className=".back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>
            </Form>
          </CardBody>,
        ],
        "className": "animated fadeIn",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h4>
              <strong>
                Book Promo - 3 of 3
              </strong>
            </h4>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Book Promo - 3 of 3
              </strong>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Book Promo - 3 of 3",
              },
              "ref": null,
              "rendered": "Book Promo - 3 of 3",
              "type": "strong",
            },
            "type": "h4",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              tag="form"
            >
              <Description
                errorMessage=""
                hasError={false}
                label="Promotional Book Description (Max 2,000 characters)*"
              />
              <AuthorBio
                errorMessage=""
                hasError={false}
                label="Author Biography:*"
              />
              
              <Button
                className=".back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>
            </Form>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Description
                  errorMessage=""
                  hasError={false}
                  label="Promotional Book Description (Max 2,000 characters)*"
                />,
                <AuthorBio
                  errorMessage=""
                  hasError={false}
                  label="Author Biography:*"
                />,
                "",
                <Button
                  className=".back-button"
                  color="basic"
                  size="sm"
                  tag="button"
                >
                  Go Back
                </Button>,
                <Button
                  color="primary"
                  size="sm"
                  tag="button"
                  type="submit"
                  value="Submit"
                >
                  Submit
                </Button>,
              ],
              "onSubmit": undefined,
              "tag": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "description": undefined,
                  "errorMessage": "",
                  "hasError": false,
                  "label": "Promotional Book Description (Max 2,000 characters)*",
                  "onDescriptionChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "authorBio": undefined,
                  "errorMessage": "",
                  "hasError": false,
                  "label": "Author Biography:*",
                  "onAuthorBioChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Go Back",
                  "className": ".back-button",
                  "color": "basic",
                  "onClick": undefined,
                  "size": "sm",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Go Back",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Submit",
                  "color": "primary",
                  "size": "sm",
                  "tag": "button",
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
